### ISSUE TITLE
`tanzu package installed update ... --wait` does not wait

### ISSUE THREAD

**Issue by mamachanko** [vmware-tanzu/tanzu-framework#968]
**Bug description**

When updating a package with `tanzu package installed update ... --wait=true` the command returns and the package reconciles eventually some time after.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [x] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**

When updating a package with `tanzu package installed update ...` the command blocks until the package is updated and has reconciled successfully.

**Steps to reproduce the bug**

I built [mamachanko/tanzu-nginx](https://github.com/mamachanko/tanzu-nginx) to reproduce the issue.

```shell
tanzu imagepullsecret add dockerhub --registry index.docker.io --username <dockerhub-username> --password <dockerhub-pwd>  --export-to-all-namespaces

tanzu package repository add nginx-repo --url mamachanko/tanzu-nginx-repo:0.0.0

# Install version 0.0.1
tanzu package install nginx --package-name nginx.mamachanko.com --version 0.0.1 --wait=true --values-file <(
  cat <<EOF
replicas: 1
delay: 5
EOF
)
# This command waits until the package has reconciled.

# Update to version 0.0.2
tanzu package installed update nginx --version 0.0.2 --wait=true --values-file <(
  cat <<EOF
replicas: 2
delay: 5
html: This is version 0.0.2
EOF
)
# This command returns and a few seconds after the package begins to reconcile.
```

**Version** (include the SHA if the version is not obvious)

```
version: v0.7.0
buildDate: 2021-10-22
sha: a683575c
```

**Environment where the bug was observed (cloud, OS, etc)**

Local kind cluster with kapp-controller installed.

**Relevant Debug Output (Logs, manifests, etc)**

**Comment by maralavi** [comment]
Thanks @mamachanko for the ```asciicinema``` recording. But at second 54s it shows the status as ```ReconcileSucceeded``` with updated version ```0.0.2```. Its true that it goes into reconciling state again after that. But that is probably a PackageInstall internal change. Otherwise, Exactly the time package update was done at second 54, the reconciliation is marked as successful in the right panel.

@cppforlife Would it make sense for PackageInstall CR to change status from ```ReconcileSucceeded``` to ```Reconciling``` without any change to the resource?

**Comment by shivaani0505** [comment]
@mamachanko , the problem there is that whats being updated is not PackageInstall, but the Secret that holds values. so PackageInstall would look like it's been reconciled successfully. eventually it notices that Secret has changed, so it will try to reconcile again.This is part of process and not a bug. Remaining all issues highlighted above have been fixed by @maralavi as a part of PR: [#390](https://github.com/vmware-tanzu/tanzu-framework/pull/390)

**Comment by mamachanko** [comment]
> @mamachanko , the problem there is that whats being updated is not PackageInstall, but the Secret that holds values. so PackageInstall would look like it's been reconciled successfully. eventually it notices that Secret has changed, so it will try to reconcile again.This is part of process and not a bug, so closing this issue. Remaining all issues highlighted above have been fixed by @maralavi as a part of PR: [#390](https://github.com/vmware-tanzu/tanzu-framework/pull/390)

Thanks outlining that @shivaani0505 ! That makes sense to me

However, I am wondering if that's what the user expects or wants. 🤔  The command `tanzu package installed update ... ---wait` suggests that it will wait until the package is "updated". Now you have explained that the `PackageInstall` resource is in fact successfully updated. But as a user I would volunteer that I am interested in the package that the `PackageInstall` represents.

I would like to propose this behaviour as I find it to be more intuitive and in line with a user's needs. However, I understand there's more to it.

**Comment by skriss** [comment]
👍  to @mamachanko's comment above, the current behavior is not very intuitive for the user as the CLI is indicating things have been updated, when in fact the package workload has yet to update.
